pandas/core/generic.py

```diff
@@ -9848,13 +9869,22 @@ def describe_1d(data):
         """
 
     @Appender(_shared_docs["pct_change"] % _shared_doc_kwargs)
-    def pct_change(self, periods=1, fill_method="pad", limit=None, freq=None, **kwargs):
+    def pct_change(
+        self: FrameOrSeries,
+        periods=1,
+        fill_method="pad",
+        limit=None,
+        freq=None,
+        **kwargs,
+    ) -> FrameOrSeries:
         # TODO: Not sure if above is correct - need someone to confirm.
         axis = self._get_axis_number(kwargs.pop("axis", self._stat_axis_name))
         if fill_method is None:
             data = self
         else:
-            data = self.fillna(method=fill_method, limit=limit, axis=axis)
+            data = self._ensure_type(
```

Comment:
why wouldn't you instead type .fillna?

Reply:
.fillna is typed as ``Optional[FrameOrSeries]``. The problem here is that ``data`` is typed to ``FrameORSeries`` and it can't accept reassigning type, per mypy's rules. the alternative to the above would be:

```python
filled = self.fillna(method=fill_method, axis=axis, limit=limit)
assert isinstance(filled, type(self))
data = filled
```

which gets repetive with a code base the size of Pandas'.

Reply:
Are you saying the Optional is throwing it off or something else?

Reply:
Yes, but by design from the mypy folks: Mypy complains if a parameter has its type redefined to a different/broader type. 

So if ‘data’ has been defined as a type DataFrame, mypy will complain if we try to redefine ‘data’ to be Optional[DataFrame].

Reply:
oh I c, since we have inplace args the return type must be Optional. this is super annoying. Let's deprecate inplace! 

yeah ok i understand why you need this.